distributed/client.py

```diff
@@ -1826,6 +1826,18 @@ def list_datasets(self, **kwargs):
         """
         return self.sync(self.scheduler.publish_list, **kwargs)
 
+    def has_dataset(self, name, **kwargs):
+        """
+        Checks for named dataset available on the scheduler
+
+        See Also
+        --------
+        Client.publish_dataset
+        Client.get_dataset
+        Client.list_datasets
+        """
+        return name in self.list_datasets()
```

Comment:
Perhaps we should just leave this to the user to do on their own?

Reply:
Adjusted this to push the check into `PublishExtension`.

Though would be better great if `get_dataset` behaved a bit more like `get` for a `Mapping` instance by taking a `default` value. ATM checking for a dataset that doesn't exist even if one uses a `try...except` triggers a lengthy amount of logging about the `name` in question not being present.